scripting/get5/recording.sp

```diff
@@ -94,9 +94,84 @@ public Action Timer_FireStopRecordingEvent(Handle timer, DataPack pack) {
   Call_PushCell(event);
   Call_Finish();
   EventLogger_LogAndDeleteEvent(event);
+
+  UploadDemoToServer(demoFileName);
   return Plugin_Handled;
 }
 
+void UploadDemoToServer(const char[] demoFileName) {
+  char demoUrl[1024];
+  char demoAuthKey[1024];
+  char demoAuthValue[1024];
+
+  g_DemoUploadUrlCvar.GetString(demoUrl, sizeof(demoUrl));
+  g_DemoUploadAuthKeyCvar.GetString(demoAuthKey, sizeof(demoAuthKey));
+  g_DemoUploadAuthValueCvar.GetString(demoAuthValue, sizeof(demoAuthValue));
+
+  if (StrEqual(demoUrl, "")) {
+    LogDebug("Will not upload any demoes as upload URL is not set.");
+    return;
+  }
+
+  if (!LibraryExists("SteamWorks")) {
+    LogDebug("Cannot upload demoes to a web server without the SteamWorks extension running.");
+    return;
+  }
+
+  char formattedUrl[1024];
+  char contentFilename[1024];
+  char userAgent[1024];
+  strcopy(formattedUrl, sizeof(formattedUrl), demoUrl);
+  ReplaceString(formattedUrl, sizeof(formattedUrl), "\"", "");
+  Format(contentFilename, sizeof(contentFilename), "attachment;filename=%s", demoFileName);
+  Format(userAgent, sizeof(userAgent), "SourceMod Get5 %s+https://%s", PLUGIN_VERSION,
+         GET5_GITHUB_PAGE);
+  // Add the protocol strings if missing (only http).
+  if (StrContains(formattedUrl, "http", false) != 0) {
+    Format(formattedUrl, sizeof(formattedUrl), "http://%s", formattedUrl);
+  }
+
+  LogDebug("UploadDemoToServer: formattedUrl (SteamWorks) = %s", formattedUrl);
+  Handle demoRequest = SteamWorks_CreateHTTPRequest(k_EHTTPMethodPOST, formattedUrl);
+
+  if (demoRequest == INVALID_HANDLE) {
+    LogDebug("Failed to create request %s", formattedUrl);
+    return;
+  }
+
+  // Set the auth keys only if they are defined. If not, we can still technically POST
+  // to an end point that has no authentication.
+  if (!StrEqual(demoAuthKey, "") && !StrEqual(demoAuthValue, "")) {
+    if (!SteamWorks_SetHTTPRequestHeaderValue(demoRequest, demoAuthKey, demoAuthValue)) {
+      LogError("Failed to add auth key and token to call. Will end request here.");
+      delete demoRequest;
+      return;
+    }
+  }
+
+  if (!SteamWorks_SetHTTPRequestHeaderValue(demoRequest, "Get5-Demoname", contentFilename)) {
+    LogError("Failed to add filename content header to call. Will end request here.");
+    delete demoRequest;
+    return;
+  }
+
+  if (!SteamWorks_SetHTTPRequestUserAgentInfo(demoRequest, userAgent)) {
+    LogError("Failed to add user-agent header to call. Will end request here.");
+    delete demoRequest;
+    return;
+  }
+
+  if (!SteamWorks_SetHTTPRequestRawPostBodyFromFile(demoRequest, "application/octet-stream",
+                                                    demoFileName)) {
+    delete demoRequest;
```

Comment:
Maybe a LogError here? Like if the file suddenly does not exist or cannot be read due to file permissions or whatever.